Android crashes when terminated lifecycle

#### Environment details

- [flutter_foreground_task: ^8.17.0](https://pub.dev/packages/flutter_foreground_task)
- [google_navigation_flutter: ^0.4.0](https://pub.dev/packages/google_navigation_flutter)
- Flutter SDK's 3.29 (latest)

#### Steps to reproduce

Entry app initialize foreground then terminated the apps


#### Code example

- flutter foreground code

```
void intialize() {
FlutterForegroundTask.init();
await FlutterForegroundTask.startService();
}
```

- google navigation flutter code
(implementation in another screen)

```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';
import 'package:google_navigation_flutter/google_navigation_flutter.dart';

class NavigationSample extends StatefulWidget {
  const NavigationSample({super.key});

  @override
  State<NavigationSample> createState() => _NavigationSampleState();
}

class _NavigationSampleState extends State<NavigationSample> {
  GoogleNavigationViewController? _navigationViewController;
  bool _navigationSessionInitialized = false;

  @override
  void initState() {
    super.initState();
    _initializeNavigationSession();
  }

  Future<void> _initializeNavigationSession() async {
    print('_initializeNavigationSession:${GoogleMapsNavigator.simulator}');
    if (!await GoogleMapsNavigator.areTermsAccepted()) {
      await GoogleMapsNavigator.showTermsAndConditionsDialog(
        'Example title',
        'Example company',
      );
    }
    // Note: make sure user has also granted location permissions before starting navigation session.
    await GoogleMapsNavigator.initializeNavigationSession();
    setState(() {
      _navigationSessionInitialized = true;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: const Text('Google Maps Navigation Sample1')),
      body: _navigationSessionInitialized
          ? GoogleMapsNavigationView(
        onViewCreated: _onViewCreated,
        initialNavigationUIEnabledPreference: NavigationUIEnabledPreference.disabled,
        // Other view initialization settings
      )
          : const Center(child: CircularProgressIndicator()),
    );
  }

  void _onViewCreated(GoogleNavigationViewController controller) {
    _navigationViewController = controller;
    controller.setMyLocationEnabled(true);
    // Additional setup can be added here.
  }

  @override
  void dispose() {
    if (_navigationSessionInitialized) {
      GoogleMapsNavigator.cleanup();
    }
    super.dispose();
  }
}
```

#### Stack trace
```
D/SessionLifecycleClient(20186): Data Collection is disabled for all subscribers. Skipping this Event
D/AndroidRuntime(20186): Shutting down VM
E/AndroidRuntime(20186): FATAL EXCEPTION: main
E/AndroidRuntime(20186): Process: com.dash.driver.android.stg, PID: 20186
E/AndroidRuntime(20186): java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
E/AndroidRuntime(20186): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:562)
E/AndroidRuntime(20186): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
E/AndroidRuntime(20186): Caused by: java.lang.reflect.InvocationTargetException
E/AndroidRuntime(20186): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime(20186): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
E/AndroidRuntime(20186): 	... 1 more
E/AndroidRuntime(20186): Caused by: com.google.maps.flutter.navigation.FlutterError: Cannot access navigation functionality before the navigation session has been initialized.
E/AndroidRuntime(20186): 	at com.google.maps.flutter.navigation.GoogleMapsNavigationSessionManager.getNavigator(GoogleMapsNavigationSessionManager.kt:151)
E/AndroidRuntime(20186): 	at com.google.maps.flutter.navigation.GoogleMapsNavigationSessionManager.unregisterListeners(GoogleMapsNavigationSessionManager.kt:285)
E/AndroidRuntime(20186): 	at com.google.maps.flutter.navigation.GoogleMapsNavigationSessionManager.onActivityDestroyed(GoogleMapsNavigationSessionManager.kt:139)
E/AndroidRuntime(20186): 	at com.google.maps.flutter.navigation.GoogleMapsNavigationPlugin.onDetachedFromActivity(GoogleMapsNavigationPlugin.kt:77)
E/AndroidRuntime(20186): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.detachFromActivity(FlutterEngineConnectionRegistry.java:381)
E/AndroidRuntime(20186): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach(FlutterActivityAndFragmentDelegate.java:754)
E/AndroidRuntime(20186): 	at io.flutter.embedding.android.FlutterActivity.onDestroy(FlutterActivity.java:909)
E/AndroidRuntime(20186): 	at android.app.Activity.performDestroy(Activity.java:8876)
E/AndroidRuntime(20186): 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1491)
E/AndroidRuntime(20186): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5600)
E/AndroidRuntime(20186): 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5645)
E/AndroidRuntime(20186): 	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:47)
E/AndroidRuntime(20186): 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
E/AndroidRuntime(20186): 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:180)
E/AndroidRuntime(20186): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:98)
E/AndroidRuntime(20186): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
E/AndroidRuntime(20186): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/AndroidRuntime(20186): 	at android.os.Looper.loopOnce(Looper.java:205)
E/AndroidRuntime(20186): 	at android.os.Looper.loop(Looper.java:294)
E/AndroidRuntime(20186): 	at android.app.ActivityThread.main(ActivityThread.java:8177)
E/AndroidRuntime(20186): 	... 3 more
I/Process (20186): Sending signal. PID: 20186 SIG: 9
Lost connection to device.
```

Thanks!

> Same here! I get the same exception thrown when using swipe kill on my Android emulator:
> 
> ```
> E/AndroidRuntime(12864): FATAL EXCEPTION: main
> E/AndroidRuntime(12864): Process: com.aeropost.driver.dev, PID: 12864
> E/AndroidRuntime(12864): java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
> E/AndroidRuntime(12864): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:562)
> E/AndroidRuntime(12864): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
> E/AndroidRuntime(12864): Caused by: java.lang.reflect.InvocationTargetException
> E/AndroidRuntime(12864): 	at java.lang.reflect.Method.invoke(Native Method)
> E/AndroidRuntime(12864): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
> E/AndroidRuntime(12864): 	... 1 more
> E/AndroidRuntime(12864): Caused by: com.google.maps.flutter.navigation.FlutterError: Cannot access navigation functionality before the navigation session has been initialized.
> E/AndroidRuntime(12864): 	at com.google.maps.flutter.navigation.GoogleMapsNavigationSessionManager.getNavigator(GoogleMapsNavigationSessionManager.kt:157)
> E/AndroidRuntime(12864): 	at com.google.maps.flutter.navigation.GoogleMapsNavigationSessionManager.unregisterListeners(GoogleMapsNavigationSessionManager.kt:294)
> E/AndroidRuntime(12864): 	at com.google.maps.flutter.navigation.GoogleMapsNavigationSessionManager.onActivityDestroyed(GoogleMapsNavigationSessionManager.kt:145)
> E/AndroidRuntime(12864): 	at com.google.maps.flutter.navigation.GoogleMapsNavigationPlugin.onDetachedFromActivity(GoogleMapsNavigationPlugin.kt:92)
> /AndroidRuntime(12864): 	at io.flutter.embedding.engine.FlutterEngineConnectionRegistry.detachFromActivity(FlutterEngineConnectionRegistry.java:381)
> E/AndroidRuntime(12864): 	at io.flutter.embedding.android.FlutterActivityAndFragmentDelegate.onDetach(FlutterActivityAndFragmentDelegate.java:754)
> E/AndroidRuntime(12864): 	at io.flutter.embedding.android.FlutterActivity.onDestroy(FlutterActivity.java:909)
> E/AndroidRuntime(12864): 	at android.app.Activity.performDestroy(Activity.java:8876)
> E/AndroidRuntime(12864): 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1491)
> E/AndroidRuntime(12864): 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:5599)
> E/AndroidRuntime(12864): 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:5644)
> E/AndroidRuntime(12864): 	at android.app.servertransaction.DestroyActivityItem.execute(DestroyActivityItem.java:47)
> E/AndroidRuntime(12864): 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
> E/AndroidRuntime(12864): 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:180)
> E/AndroidRuntime(12864): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:98)
> E/AndroidRuntime(12864): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
> I/Process (12864): Sending signal. PID: 12864 SIG: 9
> 
> E/AndroidRuntime(12864): 	at android.os.Handler.dispatchMessage(Handler.java:106)
> E/AndroidRuntime(12864): 	at android.os.Looper.loopOnce(Looper.java:205)
> E/AndroidRuntime(12864): 	at android.os.Looper.loop(Looper.java:294)
> E/AndroidRuntime(12864): 	at android.app.ActivityThread.main(ActivityThread.java:8176)
> E/AndroidRuntime(12864): 	... 3 more
> I/Process (12864): Sending signal. PID: 12864 SIG: 9
> ```
> 
> It seems to be an issue on a real device as well, because I've seen multiple times as message that my app has some crashes when the closing the app.
> 
> I'm not using FlutterForegroundTask, just GoogleMapsNavigation SDK.
> 
> How did you solve the issue?

I've resolved this crash by calling `await GoogleMapsNavigator.initializeNavigationSession();`

--- a/android/src/main/kotlin/com/google/maps/flutter/navigation/GoogleMapsNavigationSessionManager.kt
+++ b/android/src/main/kotlin/com/google/maps/flutter/navigation/GoogleMapsNavigationSessionManager.kt
@@ -291,30 +291,32 @@ private constructor(private val navigationSessionEventApi: NavigationSessionEven
   }
 
   private fun unregisterListeners() {
-    val navigator = getNavigator()
-    if (remainingTimeOrDistanceChangedListener != null) {
-      navigator.removeRemainingTimeOrDistanceChangedListener(remainingTimeOrDistanceChangedListener)
-      remainingTimeOrDistanceChangedListener = null
-    }
-    if (arrivalListener != null) {
-      navigator.removeArrivalListener(arrivalListener)
-      arrivalListener = null
-    }
-    if (routeChangedListener != null) {
-      navigator.removeRouteChangedListener(routeChangedListener)
-      routeChangedListener = null
-    }
-    if (reroutingListener != null) {
-      navigator.removeReroutingListener(reroutingListener)
-      reroutingListener = null
-    }
-    if (trafficUpdatedListener != null) {
-      navigator.removeTrafficUpdatedListener(trafficUpdatedListener)
-      trafficUpdatedListener = null
-    }
-    if (speedingListener != null) {
-      navigator.setSpeedingListener(null)
-      speedingListener = null
+    if (isInitialized()) {
+      val navigator = getNavigator()
+      if (remainingTimeOrDistanceChangedListener != null) {
+        navigator.removeRemainingTimeOrDistanceChangedListener(remainingTimeOrDistanceChangedListener)
+        remainingTimeOrDistanceChangedListener = null
+      }
+      if (arrivalListener != null) {
+        navigator.removeArrivalListener(arrivalListener)
+        arrivalListener = null
+      }
+      if (routeChangedListener != null) {
+        navigator.removeRouteChangedListener(routeChangedListener)
+        routeChangedListener = null
+      }
+      if (reroutingListener != null) {
+        navigator.removeReroutingListener(reroutingListener)
+        reroutingListener = null
+      }
+      if (trafficUpdatedListener != null) {
+        navigator.removeTrafficUpdatedListener(trafficUpdatedListener)
+        trafficUpdatedListener = null
+      }
+      if (speedingListener != null) {
+        navigator.setSpeedingListener(null)
+        speedingListener = null
+      }
     }
     if (roadSnappedLocationListener != null) {
       disableRoadSnappedLocationUpdates()

--- a/android/src/main/kotlin/com/google/maps/flutter/navigation/GoogleMapsNavigationSessionManager.kt
+++ b/android/src/main/kotlin/com/google/maps/flutter/navigation/GoogleMapsNavigationSessionManager.kt
@@ -294,7 +294,9 @@ private constructor(private val navigationSessionEventApi: NavigationSessionEven
     if (isInitialized()) {
       val navigator = getNavigator()
       if (remainingTimeOrDistanceChangedListener != null) {
-        navigator.removeRemainingTimeOrDistanceChangedListener(remainingTimeOrDistanceChangedListener)
+        navigator.removeRemainingTimeOrDistanceChangedListener(
+          remainingTimeOrDistanceChangedListener
+        )
         remainingTimeOrDistanceChangedListener = null
       }
       if (arrivalListener != null) {

--- a/ios/google_navigation_flutter/Sources/google_navigation_flutter/messages.g.swift
+++ b/ios/google_navigation_flutter/Sources/google_navigation_flutter/messages.g.swift
@@ -1814,7 +1814,7 @@ protocol NavigationViewCreationApi {
 }
 
 /// Generated setup class from Pigeon to handle messages through the `binaryMessenger`.
-class NavigationViewCreationApiSetup {
+enum NavigationViewCreationApiSetup {
   /// The codec used by NavigationViewCreationApi.
   static var codec: FlutterStandardMessageCodec { NavigationViewCreationApiCodec.shared }
   /// Sets up an instance of `NavigationViewCreationApi` to handle messages through the
@@ -2079,7 +2079,7 @@ protocol MapViewApi {
 }
 
 /// Generated setup class from Pigeon to handle messages through the `binaryMessenger`.
-class MapViewApiSetup {
+enum MapViewApiSetup {
   /// The codec used by MapViewApi.
   static var codec: FlutterStandardMessageCodec { MapViewApiCodec.shared }
   /// Sets up an instance of `MapViewApi` to handle messages through the `binaryMessenger`.
@@ -4027,7 +4027,7 @@ protocol ImageRegistryApi {
 }
 
 /// Generated setup class from Pigeon to handle messages through the `binaryMessenger`.
-class ImageRegistryApiSetup {
+enum ImageRegistryApiSetup {
   /// The codec used by ImageRegistryApi.
   static var codec: FlutterStandardMessageCodec { ImageRegistryApiCodec.shared }
   /// Sets up an instance of `ImageRegistryApi` to handle messages through the `binaryMessenger`.
@@ -4678,7 +4678,7 @@ protocol NavigationSessionApi {
 }
 
 /// Generated setup class from Pigeon to handle messages through the `binaryMessenger`.
-class NavigationSessionApiSetup {
+enum NavigationSessionApiSetup {
   /// The codec used by NavigationSessionApi.
   static var codec: FlutterStandardMessageCodec { NavigationSessionApiCodec.shared }
   /// Sets up an instance of `NavigationSessionApi` to handle messages through the
@@ -5917,7 +5917,7 @@ protocol AutoMapViewApi {
 }
 
 /// Generated setup class from Pigeon to handle messages through the `binaryMessenger`.
-class AutoMapViewApiSetup {
+enum AutoMapViewApiSetup {
   /// The codec used by AutoMapViewApi.
   static var codec: FlutterStandardMessageCodec { AutoMapViewApiCodec.shared }
   /// Sets up an instance of `AutoMapViewApi` to handle messages through the `binaryMessenger`.
@@ -7645,7 +7645,7 @@ protocol NavigationInspector {
 }
 
 /// Generated setup class from Pigeon to handle messages through the `binaryMessenger`.
-class NavigationInspectorSetup {
+enum NavigationInspectorSetup {
   /// The codec used by NavigationInspector.
   /// Sets up an instance of `NavigationInspector` to handle messages through the `binaryMessenger`.
   static func setUp(binaryMessenger: FlutterBinaryMessenger, api: NavigationInspector?) {

